File: chapter03/hello3versomp.c
#include <stdio.h> 
#include  "omp.h"
int main(void) 
{ 
   char greetingsen[] = "Hello world!"; 
   char greetingsde[] = "Hallo Welt!";
   char greetingsfr[] = "Bonjour le monde!";
   int a,b,c;

   #pragma omp parallel sections
   {
   #pragma omp section
   	{
   	for ( a = 0; a < 10; a = a + 1 )
		{
		printf ("id = %d, %s\n", omp_get_thread_num(), greetingsen); 
   		}
	}

   #pragma omp section
   	{
	for ( b = 0; b < 10; b = b + 1 )
		{
		printf ("id = %d, %s\n", omp_get_thread_num(), greetingsde); 
		}
   	}

   #pragma omp section
   	{
	for ( c = 0; c < 10; c = c + 1 )
		{
		printf ("id = %d, %s\n", omp_get_thread_num(), greetingsfr);
		} 
   	}
    }
return 0;
}

File: chapter03/hello3versomp.f90
program hello
	implicit none 
        include "omp_lib.h"

	character(len=16) :: greetingsen
	character(len=16) :: greetingsde
	character(len=16) :: greetingsfr 
	integer :: a, b, c
	greetingsen = "Hello World! " 
	greetingsde = "Hallo Welt! "
	greetingsfr = "Bonjour le monde! "	

	!$omp parallel 
	!$omp sections

	!$omp section
	do a = 1, 10
	print *, greetingsen, "From thread: ", OMP_GET_THREAD_NUM()  
	end do

	!$omp section
	do b = 1, 10
	print *, greetingsde, "Aus dem Faden: ", OMP_GET_THREAD_NUM()
	end do

	!$omp section
	do c = 1, 10
	print *, greetingsfr, "De fil: ", OMP_GET_THREAD_NUM()
	end do

	!$omp end sections 
	!$omp end parallel
	
end program hello

